### ISSUE TITLE
Editor crashes at launch after enabling autoscan of projects folder(s)

### ISSUE THREAD

**Issue by AndreaTerenz** [eumario/godot-manager#62]
**Describe the bug**
After enabling the autoscan of the project folder(s) and closing the Manager, at the next launch it will crash 

**To Reproduce**
1. Go to Settings/Projects
2. Toggle "Enable Auto Scan"
3. Reboot Godot Manager
4. Crash at launch

**Expected behavior**
The Manager should just start

**Screenshots**
Here's the exception shown when launching from console:

```
Unhandled Exception:
System.NullReferenceException: Object reference not set to an instance of an object
  at BusyDialog.UpdateHeader (System.String text) [0x00000] in <f7b093cfd2f94b2ca92bd10d8f1454ab>:0 
  at ProjectsPanel.ScanForProjects () [0x00191] in <f7b093cfd2f94b2ca92bd10d8f1454ab>:0 
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__7_0 (System.Object state) [0x00000] in <02071a39c8a34187bdd1f58e4be38764>:0 
  at Godot.GodotSynchronizationContext.ExecutePendingContinuations () [0x0001d] in <d0b7d6b6d74e4318be3dfcff5c3b7c05>:0 
  at Godot.GodotTaskScheduler.Activate () [0x0000c] in <d0b7d6b6d74e4318be3dfcff5c3b7c05>:0 
  at (wrapper native-to-managed) Godot.GodotTaskScheduler.Activate(Godot.GodotTaskScheduler,System.Exception&)

Unhandled Exception:
System.NullReferenceException: Object reference not set to an instance of an object
  at BusyDialog.UpdateHeader (System.String text) [0x00000] in <f7b093cfd2f94b2ca92bd10d8f1454ab>:0 
  at ProjectsPanel.ScanForProjects () [0x00191] in <f7b093cfd2f94b2ca92bd10d8f1454ab>:0 
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__7_0 (System.Object state) [0x00000] in <02071a39c8a34187bdd1f58e4be38764>:0 
  at Godot.GodotSynchronizationContext.ExecutePendingContinuations () [0x0001d] in <d0b7d6b6d74e4318be3dfcff5c3b7c05>:0 
  at Godot.GodotTaskScheduler.Activate () [0x0000c] in <d0b7d6b6d74e4318be3dfcff5c3b7c05>:0 
  at (wrapper native-to-managed) Godot.GodotTaskScheduler.Activate(Godot.GodotTaskScheduler,System.Exception&)
```

**Complete the Following:**
 - OS: Manjaro Linux (kernel: 6.1.12-1-MANJARO), Ryzen 5 3600, RX 580 8GB
 - Version 0.2.1

**Additional context**
NaN

**Comment by eumario** [comment]
Can you verify that in Settings > Projects > Directories to scan, are there any listed in there for one.  And for two, if there are paths listed in there, can you verify that there are no Empty ones?  You will be able to tell if there is an Empty one, but clicking below the last entry in the list, and if it highlights an empty line, it will be an empty one, and you should be able to remove that line, save it, and check and see if Auto-Scan works after that.

**Comment by kingthrillgore** [comment]
I see this behavior as well, it can be addressed by changing the variable `EnableAutoScan` to false in central_store.json in the Godot-Manager settings directory (typically set in %APPDATA% on Windows), .local on Linux. It would be nice to know if this project is still active.

**Comment by eumario** [comment]
It is active, there is currently a big re-work of the project to switch to Godot 4, and work was progressing on a new Scraper for Tuxfamily, to be more in line in the data structure as what is used with Github.  Work will hopefully be progressing faster now on the main core project, now that everything is working as it should with the new Scraper.

**Comment by eumario** [comment]
This is fixed in commit [eeb7580](https://github.com/eumario/godot-manager/commit/eeb7580a4b2e272c04d717daf4c83e3bb8366bde).  As described in the commit message, the BusyDialog is not ready in time for the startup of the application, causing it to fail to process.  This is now fixed by Looping over a wait for IdleFrame, and checking for when BusyDialog.Header is not null.
